include/ibamr/IBFEMethod.h

```diff
@@ -765,17 +765,87 @@ class IBFEMethod : public IBStrategy
     int d_max_level_number;
     std::vector<libMesh::EquationSystems*> d_equation_systems;
 
+    /// Number of parts owned by the present object.
     const unsigned int d_num_parts = 1;
+
+    /// Number of parts owned by the present object.
     std::vector<IBTK::FEDataManager*> d_fe_data_managers;
+
+    /// Minimum ghost cell width.
     SAMRAI::hier::IntVector<NDIM> d_ghosts = 0;
-    std::vector<libMesh::System*> d_X_systems, d_U_systems, d_F_systems, d_Q_systems, d_Phi_systems;
-    std::vector<libMesh::PetscVector<double> *> d_X_current_vecs, d_X_new_vecs, d_X_half_vecs, d_X_IB_ghost_vecs;
-    std::vector<libMesh::PetscVector<double> *> d_U_current_vecs, d_U_new_vecs, d_U_half_vecs;
-    std::vector<libMesh::PetscVector<double> *> d_F_half_vecs, d_F_IB_ghost_vecs;
-    std::vector<libMesh::PetscVector<double>*> d_Q_half_vecs, d_Q_IB_ghost_vecs;
-    std::vector<libMesh::PetscVector<double>*> d_Phi_half_vecs;
-
-    bool d_fe_equation_systems_initialized = false, d_fe_data_initialized = false;
+
+    /// libMesh System objects for the displacement of each part.
+    std::vector<libMesh::System *> d_X_systems;
+
+    /// libMesh System objects for the velocity of each part.
+    std::vector<libMesh::System *> d_U_systems;
+
+    /// libMesh System objects for the elastic force (i.e., what is computed
+    /// by computeInteriorForceDensity) of each part.
+    std::vector<libMesh::System *> d_F_systems;
+
+    /// libMesh System objects for body forces of each part.
+    std::vector<libMesh::System *> d_Q_systems;
+
+    /// libMesh System objects for body stress normalization of each part.
+    std::vector<libMesh::System *> d_Phi_systems;
+
+    /// Pointers to the X (displacement) vectors owned by each system.
+    std::vector<libMesh::PetscVector<double> *> d_X_current_vecs;
+
+    /// Intermediate (i.e., only used during a time step) vectors owned by
+    /// this object.
+    std::vector<libMesh::PetscVector<double> *> d_X_new_vecs;
+
+    /// Intermediate (i.e., only used during a time step) vectors owned by
+    /// this object.
+    std::vector<libMesh::PetscVector<double> *> d_X_half_vecs;
+
+    /// Pointers to vectors (owned by this object) containing ghost X
```

Comment:
These are ghosted so that each processor has access to all of the Lagrangian DOFs that can interact with local Eulerian DOFs through the IB coupling operators.